folly/portability/OpenSSL.h

```diff
@@ -35,6 +36,16 @@ namespace ssl {
 #define FOLLY_OPENSSL_IS_110 (OPENSSL_VERSION_NUMBER >= 0x10100000L)
 #endif // !defined(OPENSSL_IS_BORINGSSL)
 
+// BoringSSL and OpenSSL 1.0.2 later with TLS extension support ALPN.
+#define FOLLY_OPENSSL_HAS_ALPN      \
+  (defined(OPENSSL_IS_BORINGSSL) || \
```

Comment:
It is undefined behavior to use `defined` in the definition of a preprocessor symbol.

This form is, if less elegant, at least defined behavior:

```
#if defined(OPENSSL_IS_BORINGSSL) || \
    (OPENSSL_VERSION_NUMBER >= 0x1000200fL && \
     !defined(OPENSSL_NO_TLSEXT))
#define FOLLY_OPENSSL_HAS_ALPN 1
#else
#define FOLLY_OPENSSL_HAS_ALPN 0
#endif
```

Same for `FOLLY_OPENSSL_HAS_SNI` below.